Docs: Remove matrix from the roadmap page

xmulligan opened the pull request:
We haven't been perfect at keeping this roadmap matrix up to date and it could provide misleading information to users and developers. I think it is best to remove it so it doesn't go out of date.

Fixes: #issue-number

```release-note
Remove outdated roadmap matrix and links to it
```

xmulligan committed:
Docs: Remove matrix from the roadmap page

We haven't been perfect at keeping this roadmap matrix up to date and it could provide misleading information to users and developers. I think it is best to remove it so it doesn't go out of date.

Signed-off-by: Bill Mulligan <billmulligan516@gmail.com>

xmulligan committed:
Remove link to Roadmap in README

Signed-off-by: Bill Mulligan <billmulligan516@gmail.com>

qmonnet reviewed (commented):
Agreed that this needs to be updated or removed, the latter option being the easiest.

One note: the resulting page, titled “Roadmap” but with no roadmap on it, looks a bit strange :). I would add a statement saying that there's no (more) public roadmap published in these docs, and maybe a link to where people can find roadmap information (or what looks closest to it) - which is likely the community meeting or asking in #development on Slack. What do you think?

xmulligan committed:
Clarify roadmap is community driven

Moved the influencing the roadmap section to the top and added to it to make it clear that their is no defined roadmap, but rather driven by the community.

Signed-off-by: Bill Mulligan <billmulligan516@gmail.com>

qmonnet reviewed (approved):
Perfect! Thanks :)
